Dockerfile

```diff
@@ -2,7 +2,7 @@ ARG PYTHON_VERSION=3.12
 
 FROM python:${PYTHON_VERSION}
 RUN apt update && apt upgrade -y \
-  && apt install curl -y \
+  && apt install curl libgdal-dev -y \
```

Comment:
I would rather depends on rasterio wheels that on source + gdal binaries for now. 

rasterio hasn't been updated for a long time and there are some known compatibilities issues with later GDAL release. 

Also I'm pretty sure that installing libgdal-dev will increase the image size quite a bit.